src/btree/bt_cursor.c

```diff
@@ -234,6 +270,17 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
      * update that's been deleted is not a valid key/value pair).
      */
     if (cbt->ins != NULL) {
+        if (WT_CURSOR_HAS_BOUNDS(&cbt->iface) && btree->type == BTREE_ROW) {
+            /* Get the insert list key. */
+            if (key == NULL) {
```

Comment:
In the case that search_near() returns cbt->compare == 0, key will not be null here. Do we still need to check the insert list here? Is it possible for cursor_row_search() to return an exact key from the insert list?

Reply:
We're coding for the general case in the function, where someone could call this function with a null key and an insert list so we should handle that.